pkg/operator/backupconfigcontroller/backupconfigcontroller.go

```diff
@@ -0,0 +1,383 @@
+		"kube-scheduler",
+		revisionMap["kubeScheduler"],
+		kubescheduler.RevisionConfigMaps,
+		kubescheduler.RevisionSecrets,
+		kubescheduler.CertConfigMaps,
+		kubescheduler.CertSecrets,
+	)
+
+	kubeControllerManagerStaticPodInstallerArgs := getStaticPodInstallerArgs(
+		"kube-controller-manager",
+		revisionMap["kubeControllerManager"],
+		kubecontrollermanager.RevisionConfigMaps,
+		kubecontrollermanager.RevisionSecrets,
+		kubecontrollermanager.CertConfigMaps,
+		kubecontrollermanager.CertSecrets,
+	)
+
+	etcdStaticPodInstallerArgs := getStaticPodInstallerArgs(
+		"etcd",
+		revisionMap["etcd"],
+		etcd.RevisionConfigMaps,
+		etcd.RevisionSecrets,
+		etcd.CertConfigMaps,
+		etcd.CertSecrets,
+	)
+
+	return strings.NewReplacer(
+		"${IMAGE}", imagePullSpec,
+		"${KS_INSTALLER_FLAGS}", kubeSchedulerStaticPodInstallerArgs,
+		"${KAS_INSTALLER_FLAGS}", kubeAPIServerStaticPodInstallerArgs,
+		"${KCM_INSTALLER_FLAGS}", kubeControllerManagerStaticPodInstallerArgs,
+		"${ETCD_INSTALLER_FLAGS}", etcdStaticPodInstallerArgs,
+	), nil
+}
+
+func getStaticPodInstallerArgs(operand, revision string, secrets, configMaps, certSecrets, certConfigMaps []revisioncontroller.RevisionResource) string {
+	args := []string{
+		fmt.Sprintf("%s- -v=%d", YamlPad, 2),
+		fmt.Sprintf("%s- --revision=%s", YamlPad, revision),
+		fmt.Sprintf("%s- --namespace=%s", YamlPad, fmt.Sprintf("openshift-%s", operand)),
+		fmt.Sprintf("%s- --pod=%s", YamlPad, fmt.Sprintf("%s-pod", operand)),
+		fmt.Sprintf("%s- --resource-dir=%s", YamlPad, HostResourceDirDir),
+		fmt.Sprintf("%s- --pod-manifest-dir=%s", YamlPad, HostPodManifestDir),
+	}
+	for _, cm := range configMaps {
+		if cm.Optional {
+			args = append(args, fmt.Sprintf("%s- --optional-configmaps=%s", YamlPad, cm.Name))
+		} else {
+			args = append(args, fmt.Sprintf("%s- --configmaps=%s", YamlPad, cm.Name))
+		}
+	}
+	for _, s := range secrets {
+		if s.Optional {
+			args = append(args, fmt.Sprintf("%s- --optional-secrets=%s", YamlPad, s.Name))
+		} else {
+			args = append(args, fmt.Sprintf("%s- --secrets=%s", YamlPad, s.Name))
+		}
+	}
+	for _, cm := range certConfigMaps {
+		if cm.Optional {
+			args = append(args, fmt.Sprintf("%s- --optional-cert-configmaps=%s", YamlPad, cm.Name))
+		} else {
+			args = append(args, fmt.Sprintf("%s- --cert-configmaps=%s", YamlPad, cm.Name))
+		}
+	}
+	for _, s := range certSecrets {
+		if s.Optional {
+			args = append(args, fmt.Sprintf("%s- --optional-cert-secrets=%s", YamlPad, s.Name))
+		} else {
+			args = append(args, fmt.Sprintf("%s- --cert-secrets=%s", YamlPad, s.Name))
+		}
+	}
+	return strings.Join(args, "\n")
+}
+
+func (c *BackupConfigController) manageBackupPod(substitutionReplacer *strings.Replacer, client coreclientv1.ConfigMapsGetter, recorder events.Recorder, operatorSpec *operatorv1.StaticPodOperatorSpec) (*corev1.ConfigMap, bool, error) {
+	podBytes := etcd_assets.MustAsset("etcd/backup-pod.yaml")
+	substitutedPodString := substitutionReplacer.Replace(string(podBytes))
```

Comment:
Why not deserialize `backup-pod.yaml` into an actual `v1.Pod` object which we can manipulate in a sane way and then re-serialize?

Reply:
just time, to be honest, I will try to harden this up